### ISSUE TITLE
Size of child is different from Frame's size

### ISSUE THREAD

**Issue by anhtu293** [Visual-Behavior/aloception-oss#324]
Hello !

At the moment, I can do append a child (Depth, Disparity, Segmentation, ...) to a Frame even if their sizes are different. I think it's better to have a warning or error when we do that because it isn't normal to have dense children with different size from Frame. That can cause problems after in pipeline when we do resize, crop, ... if we don't notice the difference in size of Frame and its childrent.

```
>>> import torch
>>> import aloscene
>>> x = torch.zeros((3, 480, 640))
>>> f = aloscene.Frame(x)
/home/ubuntu/anaconda3/envs/optical-flow-anhtu/lib/python3.8/site-packages/torch/_tensor.py:997: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  ../c10/core/TensorImpl.h:1408.)
  return super(Tensor, self).rename_(names)
>>> y = torch.zeros((1, 379, 540))
>>> d = aloscene.Depth(y)
>>> f.append_depth(d)
>>> f
tensor(
	distortion=1.0
	normalization=255
	projection=pinhole
	depth=torch.Size([1, 379, 540])
	[[[0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         ...,
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.]],

        [[0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         ...,
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.]],

        [[0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         ...,
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.]]])
>>> f.resize((378, 378))
tensor(
	distortion=1.0
	normalization=255
	projection=pinhole
	depth=torch.Size([1, 298, 319])
	[[[0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         ...,
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.]],

        [[0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         ...,
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.]],

        [[0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         ...,
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.],
         [0., 0., 0.,  ..., 0., 0., 0.]]])
```

**Comment by thibo73800** [comment]
it isn't normal to have dense children with different size. Not sure it is not normal even though this is not usual.

You say ''cause problems after in pipeline when we do resize, crop" => What problems exactly ?

**Comment by anhtu293** [comment]
They are not the problems of `aloception-oss` but the problems in some of my training pipelines.
For example, in one depth dataset that i'm working on, there are few samples whose depth labels have different size as RGB. Because `aloception-oss` accepts children with different size silently while loading frames, it takes times to debug.
